### PR TITLE
FastAPI: capture responseHeadersOnEntrySpans

### PR DESCRIPTION
**Pull request by GSVarsha**
FastAPI: capture responseHeadersOnEntrySpans

Signed-off-by: Varsha GS <varsha.gs@ibm.com>

### CODE REVIEW

File: instana/instrumentation/asgi.py [hunk 1]
```diff
@@ -21,11 +21,12 @@ def __init__(self, app):
 
     def _extract_custom_headers(self, span, headers):
         try:
-            for custom_header in agent.options.extra_http_headers:
-                # Headers are in the following format: b'x-header-1'
-                for header_pair in headers:
-                    if header_pair[0].decode('utf-8').lower() == custom_header.lower():
-                        span.set_tag("http.header.%s" % custom_header, header_pair[1].decode('utf-8'))
+            if agent.options.extra_http_headers is not None:
```
Comment (Ferenc-) [inline instana/instrumentation/asgi.py:24]: This check should be inverted for an early return and should go outside of the try block.

File: tests/apps/fastapi_app/app.py [hunk 2]
```diff
@@ -24,6 +24,11 @@ async def root():
 async def user(user_id):
     return {"user": user_id}
 
+@fastapi_server.get("/response_headers")
+async def response_headers():
+    headers = {'X-Capture-This-Too': 'this too'}
```
Comment (Ferenc-) [inline tests/apps/fastapi_app/app.py:29]: I think it is safer to test with multiple headers at a time.

File: tests/frameworks/test_fastapi.py [hunk 3]
```diff
@@ -366,3 +366,53 @@ def test_custom_header_capture(server):
...
+
+
+def test_response_header_capture(server):
+
+     # The background FastAPI server is pre-configured with custom headers to capture
+    
+    with tracer.start_active_span('test'):
+        result = requests.get(testenv["fastapi_server"] + '/response_headers')
+
+    assert result.status_code == 200
```
Comment (Ferenc-) [inline tests/frameworks/test_fastapi.py:378]: This TS was unfortunately also full with non-unit test assert, but perhaps we could use `self.assertEqual` and further below.

**Review by Ferenc- — CHANGES_REQUESTED** [review]
Looking good. I have requested a couple of small changes.

File: instana/instrumentation/asgi.py (hunk 1 above)
Comment (pvital) [reply instana/instrumentation/asgi.py:24]: +1 for Ferenc's comment. You can follow the same as you did for Django's PR.

**Review by pvital — APPROVED** [review]
Besides the one comment, It looks good to me.

**Review by Ferenc- — APPROVED** [review]
Approved.

**Pull request merged**